Modules/_ctypes/callproc.c

```diff
@@ -1014,7 +1014,10 @@ void _ctypes_extend_error(PyObject *exc_class, const char *fmt, ...)
 
     PyErr_Fetch(&tp, &v, &tb);
     PyErr_NormalizeException(&tp, &v, &tb);
-    cls_str = PyObject_Str(tp);
+    if (PyType_Check(tp))
+        cls_str = PyType_GetName((PyTypeObject *)tp);
```

Comment:
```suggestion
        cls_str = ((PyTypeObject *)tp)->tp_name;
```

Reply:
Oops that was wrong, tp_name is a char*, not a PyObject*.

Reply:
@kamilturek would you be willing to do a manual backport to the 3.10 branch? We'll need a slightly different approach.

Reply:
@JelleZijlstra Sure thing, will do 👍

Reply:
The backport with a different approach: https://github.com/python/cpython/pull/99452.